client/gutenboarding/index.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import { createSite, main, redirectIfNotEnabled } from './controller';
+import { makeLayout, render as clientRender } from 'controller';
+import { hideMasterbar as hideMasterbarAction } from 'state/ui/actions';
+
+function hideMasterbar( context: PageJS.Context, next ) {
+	context.store.dispatch( hideMasterbarAction() );
+	next();
+}
+
+export default function() {
+	page( '/gutenboarding', redirectIfNotEnabled, hideMasterbar, main, makeLayout, clientRender );
+	page(
+		'/gutenboarding/onboard',
+		redirectIfNotEnabled,
+		hideMasterbar,
```

Comment:
Minor, but this middleware might not be necessary, I think :thinking: The crux is in `makeLayout,` which really just creates a `<Layout />` root component in `context.layout`, with `primary` and `secondary` taken as props from routing `context`, https://github.com/Automattic/wp-calypso/blob/afb2a58683accd4fced840f19c94fe09f29f0bef/client/controller/shared.js#L21-L28

I'll try changing the `main` middleware to write to `context.layout` directly (which is kind of how I always hoped we'd write our middleware for sections without sidebars -- `makeLayout` was meant more as a stop-gap compat thingy for the single-tree migration.)

Reply:
Uh, nvm, what I'm suggesting might not apply to the masterbar (but only the sidebar) :roll_eyes: